src/app/api/chapter/route.ts

```diff
@@ -0,0 +1,93 @@
+import { ChapterSearchParams, IChapter, RecursivePartial, SearchRequestBody } from "@/common_utils/types";
+import { deleteChapter, getChapterByName, getChaptersFiltered, updateChapter } from "@server/mongodb/actions/Chapter";
+import APIWrapper from "@server/utils/APIWrapper";
+
+
+
+export const GET = APIWrapper({
+    config: { requireVolunteer: true },
+    handler: async (req) => {
+        const { searchParams } = new URL(req.url);
+        const name = searchParams.get("name")
+
+        if (!name) {
+            throw new Error("Chapter Name is missing")
+        }
+
+        const chapter = await getChapterByName(name);
+        return chapter
+    }
+})
+
+
+
+type PatchReq = {
+    name: string;
+    newFields: RecursivePartial<IChapter>;
+  };
+
+export const PATCH = APIWrapper({
+    config: { requireAdmin: true },
+    handler: async (req) => {
```

Comment:
should i add updateCookie on these requests?

Reply:
no need to, only used for login and auth.